include/mbedtls/config.h

```diff
@@ -56,6 +56,22 @@
 #define MBEDTLS_HAVE_ASM
 
 /**
+ * \def MBEDTLS_NO_INT64_DIVISION
+ *
+ * The platform lacks support for 64-bit division.
+ *
+ * Used in:
+ *      include/mbedtls/bignum.h
+ *      library/bignum.c
+ *
+ * Uncomment to prevent the use of 64-bit division, even if a 64-bit type
+ * is available. This allows building for a platform that lacks a hardware
+ * 64-bit division, if linking with a software implementation of division
+ * is not desired. Note that 32-bit division is required.
+ */
+//#define MBEDTLS_NO_INT64_DIVISION
```

Comment:
I agree with this fix, but I disagree with this new compile-time option as it is only for 64-bit software arithmetic operations. What is stopping anyone from requesting that we disable 128-bit types when using a 64-bit machine?

As an alternative, I would prefer to change this compile time option so that it disables software arithmetic operations with double the machine's word size even if such type is available. Then the code in `include/mbedtls/bignum.h` could have to be modified so that 128-bit types are not used when compiling 64-bits. @gilles-peskine-arm what do you think?